proto/proto-backplane-grpc/src/main/proto/deephaven/proto/ticket.proto

```diff
@@ -23,4 +23,13 @@ option optimize_for = SPEED;
  */
 message Ticket {
     bytes ticket = 1;
+}
+
+message TypedTicket {
+    bytes ticket = 1;
+
+    /*
+     * The type. Its absence means that it is not known, not that the server chose to not set it.
+     */
+    optional string type = 2;
```

Comment:
optional isn't necessary here, empty is "wrong", we don't support emptystring in other places either right? at least earlier revisions didn't...

Reply:
I hate (ab)using the empty string given the more explicit protobuf support for optional; that said, we could also have a meaningful "type" like "Unknown" if we think that would make things clear-er?

Reply:
I'm also "ok" with dropping optional and using empty string.

Reply:
Latest commit I switched over - essentially, the difference theoretically between:
```java
x.hasType()
```
vs
```java
!x.getType().isEmpty()
```

Reply:
Note that there is no such thing as empty string in protobuf, the tag is just absent. The advantage of "optional" is that now "null" and "empty" are two cases, rather than one, so you are actually _adding_ support for empty string as an explicit thing which is different than null, rather than removing.

With a non-optional, the default string is non-null, and in protobuf's eyes, that is the absence of a string, until you explicitly make it optional. Even then, you can't get null! You just get a `.hasType()` which now will return false, even when `.getType()` returns `""`.